.ci/util.sh

```diff
@@ -31,9 +31,11 @@ function checkout_from {
   mkdir -p .ci-temp
   cd .ci-temp
   if [ -d "$PROJECT" ]; then
-    echo "Target project $PROJECT is already cloned, latest changes will be fetched"
+    echo "Target project $PROJECT is already cloned, latest changes will be fetched and reset"
     cd "$PROJECT"
     git fetch
+    git reset --hard HEAD
+    git clean -f -d
```

Comment:
https://git-scm.com/docs/git-clean
> Don’t use the standard ignore rules (see [gitignore[5]](https://git-scm.com/docs/gitignore)), but still use the ignore rules given with -e options from the command line.

To me this doesn't seem necessary since we should only have maven artifacts and we use clean on the builds. CI should stop if any command fails, so it will present issues there.

Reply:
This was more of a suggestion, to make sure that we basically have a freshly cloned repo for local execution. In CI, it doesn't matter since we have a fresh container each time and only cache dependencies in some cases.